test/task-checker.js

```diff
@@ -46,51 +48,53 @@ describe('This should enforce any rules we want to exist for our tasks', () => {
   tasksToTest.map(taskObject => {
     let taskName = taskObject.taskName;
     let task = taskObject.task;
-
-    it('should only have supported methods', () => {
-      let supportedMethods = ['build', 'watch'];
-      let invalidKeys = Object.keys(task).filter(key => {
-        if (supportedMethods.indexOf(key) === -1) {
-          return key;
+    describe(taskName, () => {
+      it('should only have supported methods', () => {
+        let supportedMethods = ['build', 'watch'];
+        let invalidKeys = Object.keys(task).filter(key => {
+          if (supportedMethods.indexOf(key) === -1) {
+            return key;
+          }
+        });
+
+        if (invalidKeys.length > 0) {
+          let keysString = invalidKeys.join(',');
+          throw new Error(`There are unexpected methods ` +
+            `in the "${taskName}" task: [ ${keysString} ]`);
         }
       });
 
-      if (invalidKeys.length > 0) {
-        let keysString = invalidKeys.join(',');
-        throw new Error(`There are unexpected methods ` +
-          `in the "${taskName}" task: [ ${keysString} ]`);
-      }
-    });
-
-    it('should have all required methods', () => {
-      let requiredMethods = ['build'];
-      let taskKeys = Object.keys(task);
-      let missingMethods = requiredMethods.filter(key => {
-        if (taskKeys.indexOf(key) === -1) {
-          return key;
+      it('should have all required methods', () => {
+        let requiredMethods = ['build'];
+        let taskKeys = Object.keys(task);
+        let missingMethods = requiredMethods.filter(key => {
+          if (taskKeys.indexOf(key) === -1) {
+            return key;
+          }
+        });
+
+        if (missingMethods.length > 0) {
+          let keysString = missingMethods.join(',');
+          throw new Error(`There are missing required methods ` +
+            `in the "${taskName}" task: [ ${keysString} ]`);
         }
       });
 
-      if (missingMethods.length > 0) {
-        let keysString = missingMethods.join(',');
-        throw new Error(`There are missing required methods ` +
-          `in the "${taskName}" task: [ ${keysString} ]`);
-      }
-    });
-
-    it('should build cleanly', done => {
-      GLOBAL.config = {
-        env: 'prod',
-        src: VALID_TEST_FILES,
-        dest: TEST_OUTPUT_PATH
-      };
-
-      task.build().on('end', () => {
-        // Check output exists
-        var outputFiles = fs.readdirSync('test/output');
-        outputFiles.should.have.length.above(0);
-
-        done();
+      it('should build cleanly', function(done) {
+        this.timeout(10000);
```

Comment:
Nah keep the timeout in place but 3 seconds for that single file is a little alarming tbh :-/